jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -30,18 +36,20 @@ def activate(self):
         self.viewer.state.add_callback('y_min', self.on_limits_change)
         self.viewer.state.add_callback('y_max', self.on_limits_change)
 
+        # NOTE: For Imviz only.
         # Set the reference data in other viewers to be the same as the current viewer.
         # If adding the data to the viewer, make sure it is not actually shown since the
         # user didn't request it.
-        for viewer in self.viewer.session.application.viewers:
-            if viewer is not self.viewer:
-                if self.viewer.state.reference_data not in viewer.state.layers_data:
-                    viewer.add_data(self.viewer.state.reference_data)
-                    for layer in viewer.state.layers:
-                        if layer.layer is self.viewer.state.reference_data:
-                            layer.visible = False
-                            break
-                viewer.state.reference_data = self.viewer.state.reference_data
+        if self.viewer.jdaviz_app.config == 'imviz':
+            for viewer in self._iter_image_viewers():
+                if viewer is not self.viewer:
+                    if self.viewer.state.reference_data not in viewer.state.layers_data:
+                        viewer.add_data(self.viewer.state.reference_data)
+                        for layer in viewer.state.layers:
+                            if layer.layer is self.viewer.state.reference_data:
+                                layer.visible = False
+                                break
```

Comment:
was this not covered in CI before as well and just being triggered as failing this CI because of the extra indentation?  Is there an easy way to add coverage for this?

Reply:
Tests added!